File: Cevsen/MainActivity.java
package com.serifyasargmail.cevsen;

import android.content.Intent;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getSupportActionBar().setTitle("Cevşen ve Dualar");
        Button btnCevsen=findViewById(R.id.btnCevsen);
        btnCevsen.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                String s = "cevsen";
                Intent i = new Intent(getApplicationContext(), CevsenActivity.class);
                i.putExtra("dua",s);
                startActivity(i);

            }
        });

        Button btnBedir=findViewById(R.id.btnBedir);
        btnBedir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                String s = "abedir";
                Intent i = new Intent(getApplicationContext(), CevsenActivity.class);
                i.putExtra("dua",s);
                startActivity(i);

            }
        });


        Button btnYasin=findViewById(R.id.btnYasin);
        btnYasin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                String s = "yasin";
                Intent i = new Intent(getApplicationContext(), CevsenActivity.class);
                i.putExtra("dua",s);
                startActivity(i);

            }
        });


        Button btnSabah=findViewById(R.id.btnSabah);
        btnSabah.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                String s = "sabah";
                Intent i = new Intent(getApplicationContext(), CevsenActivity.class);
                i.putExtra("dua",s);
                startActivity(i);

            }
        });


        Button btnAksam=findViewById(R.id.btnAksam);
        btnAksam.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                String s = "aksam";
                Intent i = new Intent(getApplicationContext(), CevsenActivity.class);
                i.putExtra("dua",s);
                startActivity(i);

            }
        });

        Button btnFetih=findViewById(R.id.btnFetih);
        btnFetih.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                String s = "fetih";
                Intent i = new Intent(getApplicationContext(), CevsenActivity.class);
                i.putExtra("dua",s);
                startActivity(i);

            }
        });


    }
}

File: Quiz/ui/MainActivity.kt
package com.serifyasar.quiz.ui

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.google.android.material.chip.Chip
import com.serifyasar.quiz.R
import com.serifyasar.quiz.api.ApiClient
import com.serifyasar.quiz.api.ApiQuestion
import com.serifyasar.quiz.data.Question
import com.serifyasar.quiz.data.Repository
import com.serifyasar.quiz.data.SorguDao
import com.serifyasar.quiz.data.Veritabani
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Response
import java.util.*

class MainActivity : AppCompatActivity() {


    private lateinit var viewModel: SoruViewModel
    private lateinit var sorular : MutableList<Question>
    private var s=0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


      //  ilkSoruAl()
        viewModel=ViewModelProviders.of(this).get(SoruViewModel::class.java)

        viewModel.sorular.observe(this, Observer {
            if(it.isNotEmpty()) {
                sorular= it as MutableList<Question>

                soruCek()

                btnNext.setOnClickListener {
                    val secilen=findViewById<Chip>(chipGrup.checkedChipId)
                    if(secilen.text==sorular[s].cevap)
                        viewModel.d++
                    else
                        viewModel.y++

                    viewModel.index++
                    if (viewModel.index<10) {
                        sorular.removeAt(s)
                        soruCek()

                    }else{

                        val intent:Intent = Intent(this,SonucActivity::class.java)
                        intent.putExtra("dogru",viewModel.d)
                        intent.putExtra("yanlis",viewModel.y)

                        viewModel.d=0
                        viewModel.y=0
                        viewModel.index=0

                        chipGrup.clearCheck()
                        startActivity(intent)

                    }


                }
            }
        })


    }



    fun soruCek(){

        s=Random().nextInt(sorular.size-1)
        var liste= listOf<String>(sorular[s].cevap_a,sorular[s].cevap_b, sorular[s].cevap_c,sorular[s].cevap_d)
        liste= liste.shuffled()
        chipGrup.clearCheck()
        tvSoru.text=sorular[s].soru
        chipA.text=liste.get(0)
        chipB.text=liste.get(1)
        chipC.text=liste.get(2)
        chipD.text=liste.get(3)

    }


}
